File: 2003/IntroToVBNETPart2/Presentation/Overload.cs
using System ;

public class DisplayIt
{
    public void Display ( char theChar )
    {
        Console.WriteLine ( "Display: theChar      : {0}", theChar ) ;
    }
    public void Display ( int theInteger )
    {
        Console.WriteLine ( "Display: theInteger   : {0}", theInteger ) ;
    }
    public void Display ( string theString )
    {
        Console.WriteLine ( "Display: theString    : {0}", theString ) ;
    }
}

public class Test
{
    static void Main ()
    {
        DisplayIt d = new DisplayIt () ;
        d.Display ( (char) 86 ) ;
        d.Display ( 'B' ) ;
        d.Display ( '.' ) ;
        d.Display ( "NET" ) ;
        d.Display ( 12345 ) ;
    }
 }

File: 2003/IntroToVBNETPart2/Presentation/Overload.vb
Imports System
Imports Microsoft.VisualBasic

Public Class DisplayIt
   Overloads Public Sub Display ( theChar As Char )
       Console.WriteLine ( "Display: theChar      : {0}", theChar )
   End Sub
   Overloads Public Sub Display ( theInteger As Integer )
        Console.WriteLine ( "Display: theInteger   : {0}", theInteger )
   End Sub
   Overloads Public Sub Display ( theString As String )
        Console.WriteLine ( "Display: theString    : {0}", theString )
   End Sub
End Class

Public Class Test
    Shared Sub Main()
        Dim d As New DisplayIt
        d.Display ( ChrW ( 86 ) )
        d.Display ( "B"C )
        d.Display ( "."C )
        d.Display ( "NET" ) 
        d.Display ( 12345 )
    End Sub
End Class